src/layers/conv.jl

```diff
@@ -1065,4 +1065,101 @@ function (l::MEGNetConv)(g::GNNGraph, x::AbstractMatrix, e::AbstractMatrix)
     return x̄, ē
 end
 
+@doc raw"""
+    GMMConv(in => out, K, e_dim, σ=identity; [init, bias])
+Graph mixture model convolution layer from the paper [Geometric deep learning on graphs and manifolds using mixture model CNNs](https://arxiv.org/abs/1611.08402)
+Performs the operation
+```math
+\mathbf{x}_i' = \frac{1}{|N(i)|} \sum_{j\in N(i)}\frac{1}{K}\sum_{k=1}^k \mathbf{w}_k(\mathbf{e}_{j\to i}) \odot \Theta_k \mathbf{x}_j
+```
+where
+```math
+w^a_{k}(e^a) = \exp(\frac{-1}{2}(e^a - \mu^a_k)^T \Sigma^a_k^{-1}(e^a - \mu^a_k))
+```
+$\Theta_k$, $\mu^a_k$, $\Sigma^a_k^{-1}$ are learnable parameters.
 
+The input to the layer is a node feature array 'X' of size `(num_features, num_nodes)` and
+edge pseudo-cordinate array 'U' of size `(num_features, num_edges)`
+# Arguments 
+- `in`: Number of input features.
+- `out`: Number of output features.
+- `K` : Number of kernels. Default `1`.
+- `e_dim` : Dimensionality of pseudo coordinates. Has to correspond to the edge features dimension in the forward pass.
+- `σ`: Activation function. Default `identity`.
+- `bias`: Add learnable bias. Default `true`.
+- `init`: Weights' initializer. Default `glorot_uniform`.
+
+#Examples
+
+```julia
+# create data
+s = [1,1,2,3]
+t = [2,3,1,1]
+g = GNNGraph(s,t)
+in_feature, out_feature, K, e_dim = 4, 7, 8, 10
+x = randn(in_feature, g.num_nodes)
+e = randn(e_dim, g.num_edges)
+
+# create layer
+l = GMMConv(in_feature=>out_feature, K, e_dim)
+
+# forward pass
+l(g, x, e)
+```
+"""
+struct GMMConv{A<:AbstractMatrix, B, F} <:GNNLayer
+    mu::A
+    sigma_inv::A
+    bias::B
+    σ::F
+    ch::Pair{Int, Int}
+    K::Int
+    e_dim::Int
+    dense_x::Dense
+end
+
+@functor GMMConv
+
+function GMMConv(ch::Pair{Int, Int}, 
+                σ=identity;
+                K::Int=1,
+                e_dim::Int=1,
+                init=Flux.glorot_uniform,
+                bias::Bool=true)
+    in, out = ch
+    mu = init(K, e_dim)
+    sigma_inv = init(K, e_dim)
+    b = bias ? Flux.create_bias(ones(out), true) : false
+    dense_x = Dense(in, out*K, bias=false)
+    GMMConv(mu, sigma_inv, b, σ, ch, K, e_dim, dense_x)
+end
+
+function (l::GMMConv)(g::GNNGraph, x::AbstractMatrix, u::AbstractMatrix)
+
+    @assert (l.e_dim == size(u)[1] && g.num_edges == size(u)[2]) "Pseudo-cordinate dim $(size(u)) does not match (e_dim=$(e_dim),num_edge=$(g.num_edges))"
+
+    num_edges = g.num_edges
+    d = degree(g, dir=:in)
+    u = reshape(u, (l.e_dim, 1, num_edges))
+    mu = reshape(l.mu, (l.e_dim, l.K, 1))
+    
+    e = -0.5*(u.-mu).^2
+    e = e .* ((reshape(l.sigma_inv, (l.e_dim, l.K, 1)).^2) )
```

Comment:
```suggestion
    e = e .* reshape(l.sigma_inv, (l.e_dim, l.K, 1))
```
No need to square since `sigma_inv`  is an inverse covariance